src/Bedrock.Framework.Experimental/Protocols/Tls/BulkCiphers/BulkCipher.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Text;
+
+namespace Bedrock.Framework.Experimental.Protocols.Tls.BulkCiphers
+{
+    internal abstract class BulkCipher : IDisposable
```

Comment:
What is this type for?

Reply:
There is a difference in how TLS 1.2 and TLS 1.3 for instance encrypt each record (different additional info, different IV calculations for each, different headers and footers for the encryption frame) so this way the up level protocol handlers can abstract that detail and not care. They just have a Decrypt/Encrypt. I will upload the TLS 1.2 and TLS 1.3 implementation of each of these next.